src/vector/_backends/numpy_.py

```diff
@@ -474,8 +472,9 @@ def __array_ufunc__(
         ):
             result = inputs[0].add(inputs[1])
             for output in outputs:
+                assert output.dtype.names is not None
                 for name in output.dtype.names:
-                    output[name] = result[name]
+                    output[name] = result[name]  # type: ignore
```

Comment:
Just punted on this one. VectorNumpy doesn't support assignment, and VectorProtocol doesn't support getitem.